tests/unit/backend/io/test_metadata.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+"""
+Unit tests for backend metadata auxiliary class.
+"""
+import os
+import pytest
+import numpy as np
+import tifffile
+from imars3d.backend.io.metadata import MetaData
+from imars3d.backend.io.metadata import _extract_metadata_from_tiff
+
+
+@pytest.fixture(scope="module")
+def test_data():
+    # create testing tiff images
+    data = np.ones((3, 3))
+    #
+    ext_tags_ct = [
+        (65026, "s", 0, "ManufacturerStr:Test", True),
+        (65027, "s", 0, "ExposureTime:70.000000", True),
+        (65068, "s", 0, "MotSlitHR.RBV:10.000000", True),
+        (65070, "s", 0, "MotSlitHL.RBV:20.000000", True),
+        (65066, "s", 0, "MotSlitVT.RBV:10.000000", True),
+        (65068, "s", 0, "MotSlitHR.RBV:10.000000", True),
+    ]
+    ext_tags_dc = [
+        (65026, "s", 0, "ManufacturerStr:Test", True),
+        (65027, "s", 0, "ExposureTime:70.000000", True),
+    ]
+    ext_tags_ct_alt = [
+        (65026, "s", 0, "ManufacturerStr:Test", True),
+        (65027, "s", 0, "ExposureTime:71.000000", True),
+        (65068, "s", 0, "MotSlitHR.RBV:11.000000", True),
+        (65070, "s", 0, "MotSlitHL.RBV:21.000000", True),
+        (65066, "s", 0, "MotSlitVT.RBV:11.000000", True),
+        (65068, "s", 0, "MotSlitHR.RBV:11.000000", True),
+    ]
+    # write testing data
```

Comment:
This should use the [pytest `tmpdir` fixture](https://docs.pytest.org/en/6.2.x/tmpdir.html). That will give a reliable place to write to and will remove the directory and its contents once the test is complete, independent of whether it passed.

Reply:
switch to use pytest tmp fixture as suggested